class MyLinkedList {
    class Node {
        int val;
        Node next;
      

        Node(int val) {
            this.val = val;
            this.next = null;
        }
    }
   
     Node head;
    public MyLinkedList() {
        head= null;
        
    }
    
    public int get(int index) {
         Node temp = head;

        for (int i = 0; i < index; i++) {

            if (temp == null) {
                return -1;
            }

            temp = temp.next;
        }

        if (temp == null) {
            return -1;
        }

        return temp.val;
        
    }
    
    public void addAtHead(int val) {
         Node newNode = new Node(val);

        newNode.next = head;
        head = newNode;
        return ;

    }
    
    public void addAtTail(int val) {
        Node newNode = new Node(val);

        if (head == null) {
            head = newNode;
            return;
        }

        Node temp = head;

        while (temp.next != null) {
            temp = temp.next;
        }

        temp.next = newNode;
        
    }
    
    public void addAtIndex(int index, int val) {
         Node newNode = new Node(val);

    // Insert at head
    if (index == 0) {
        newNode.next = head;
        head = newNode;
        return;
    }

    Node temp = head;

    // Reach the node just before the index
    for (int i = 1; i < index; i++) {
        temp = temp.next;
    }

    // Insert the new node
    newNode.next = temp.next;
    temp.next = newNode;

        
    }
    
    public void deleteAtIndex(int index) {
        // Empty list
        if (head == null) {
            return;
        }

        // Delete head
        if (index == 0) {
            head = head.next;
            return;
        }

        Node temp = head;

        // Reach node before the index
        for (int i = 1; i < index; i++) {

            if (temp == null || temp.next == null) {
                return;
            }

            temp = temp.next;
        }

        // Delete node
        if (temp.next != null) {
            temp.next = temp.next.next;
        }    
        
    }
}

/**
 * Your MyLinkedList object will be instantiated and called as such:
 * MyLinkedList obj = new MyLinkedList();
 * int param_1 = obj.get(index);
 * obj.addAtHead(val);
 * obj.addAtTail(val);
 * obj.addAtIndex(index,val);
 * obj.deleteAtIndex(index);
 */